pkg/controller/controller_instance.go

```diff
@@ -203,10 +210,34 @@ func (c *controller) reconcileInstanceDelete(instance *v1alpha1.Instance) error
 	return nil
 }
 
+// isInstanceFailed returns whether the instance has a failed condition with
+// status true.
+func isInstanceFailed(instance *v1alpha1.Instance) bool {
+	for _, condition := range instance.Status.Conditions {
+		if condition.Type == v1alpha1.InstanceConditionFailed && condition.Status == v1alpha1.ConditionTrue {
+			return true
+		}
+	}
+
+	return false
+}
+
 // reconcileInstance is the control-loop for reconciling Instances. An
 // error is returned to indicate that the binding has not been fully
 // processed and should be resubmitted at a later time.
 func (c *controller) reconcileInstance(instance *v1alpha1.Instance) error {
+	// Currently, we only set a failure condition if the initial provision
+	// call fails, so if that condition is set, there is no more work to do.
+	// We will need to reevaluate this logic as we make any changes to capture
```

Comment:
What about the orphan mitigation? Shouldn't we delete the potentially created resource before giving up or retrying to provision again?